comp/dogstatsd/server/batch.go

```diff
@@ -96,6 +137,13 @@ func newBatcher(demux aggregator.DemultiplexerWithAggregator) *batcher {
 	samplesWithTs := demux.GetMetricSamplePool().GetBatch()
 	samplesWithTsCount := 0
 
+	shardGenerator := &shardKeyGeneratorIsolation{
```

Comment:
I don't have a strong opinion here, but are you sure it's alright to change the sharding strategy here unconditionally?

Reply:
This is a mistake, I'll fix